diff --git a/.gitignore b/.gitignore
index 755ec9c2..863bd3d5 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1,5 @@
 scratch.txt
+.direnv
+flake.nix
+flake.lock
+.envrc
\ No newline at end of file
diff --git a/cmd/gluetun/main.go b/cmd/gluetun/main.go
index c5b47126..a2a5035b 100644
--- a/cmd/gluetun/main.go
+++ b/cmd/gluetun/main.go
@@ -289,7 +289,13 @@ func _main(ctx context.Context, buildInfo models.BuildInformation,
 	if err != nil {
 		return err
 	}
-	err = os.MkdirAll("/gluetun", permission)
+
+	home := os.Getenv("GLUETUN_HOME")
+	if home == "" {
+		home = "/gluetun"
+	}
+
+	err = os.MkdirAll(home, permission)
 	if err != nil {
 		return err
 	}
diff --git a/internal/cli/formatservers.go b/internal/cli/formatservers.go
index 254c0451..92bad849 100644
--- a/internal/cli/formatservers.go
+++ b/internal/cli/formatservers.go
@@ -80,8 +80,13 @@ func (c *CLI) FormatServers(args []string) error {
 		}
 	}
 
+	storagePath := os.Getenv("STORAGE_FILEPATH")
+	if storagePath == "" {
+		storagePath = constants.ServersData
+	}
+
 	logger := newNoopLogger()
-	storage, err := storage.New(logger, constants.ServersData)
+	storage, err := storage.New(logger, storagePath)
 	if err != nil {
 		return fmt.Errorf("creating servers storage: %w", err)
 	}
diff --git a/internal/cli/openvpnconfig.go b/internal/cli/openvpnconfig.go
index 6987d496..1dcf3d82 100644
--- a/internal/cli/openvpnconfig.go
+++ b/internal/cli/openvpnconfig.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"net/http"
 	"net/netip"
+	"os"
 	"strings"
 	"time"
 
@@ -46,7 +47,12 @@ type IPv6Checker interface {
 func (c *CLI) OpenvpnConfig(logger OpenvpnConfigLogger, reader *reader.Reader,
 	ipv6Checker IPv6Checker,
 ) error {
-	storage, err := storage.New(logger, constants.ServersData)
+	storagePath := os.Getenv("STORAGE_FILEPATH")
+	if storagePath == "" {
+		storagePath = constants.ServersData
+	}
+
+	storage, err := storage.New(logger, storagePath)
 	if err != nil {
 		return err
 	}
diff --git a/internal/cli/update.go b/internal/cli/update.go
index 9e684490..2a5cc88d 100644
--- a/internal/cli/update.go
+++ b/internal/cli/update.go
@@ -6,6 +6,7 @@ import (
 	"flag"
 	"fmt"
 	"net/http"
+	"os"
 	"strings"
 	"time"
 
@@ -71,7 +72,12 @@ func (c *CLI) Update(ctx context.Context, args []string, logger UpdaterLogger) e
 		return fmt.Errorf("options validation failed: %w", err)
 	}
 
-	storage, err := storage.New(logger, constants.ServersData)
+	storagePath := os.Getenv("STORAGE_FILEPATH")
+	if storagePath == "" {
+		storagePath = constants.ServersData
+	}
+
+	storage, err := storage.New(logger, storagePath)
 	if err != nil {
 		return fmt.Errorf("creating servers storage: %w", err)
 	}
diff --git a/internal/configuration/sources/files/reader.go b/internal/configuration/sources/files/reader.go
index 477e39df..13f27d28 100644
--- a/internal/configuration/sources/files/reader.go
+++ b/internal/configuration/sources/files/reader.go
@@ -25,8 +25,13 @@ func New(warner Warner) (source *Source) {
 		environ[split[0]] = split[1]
 	}
 
+	home := environ["GLUETUN_HOME"]
+	if home == "" {
+		home = "/gluetun"
+	}
+
 	return &Source{
-		rootDirectory: "/gluetun",
+		rootDirectory: home,
 		environ:       environ,
 		warner:        warner,
 	}
